### PR TITLE
Upgrade ReScript to v11

### PR DESCRIPTION
**Pull request by jihchi**
See [the post](https://rescript-lang.org/blog/improving-interop) for more details about the upcoming ReScript 11.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "@testing-library/react": "^14.0.0",
     "@vitejs/plugin-react": "^4.0.0",
     "jsdom": "^22.0.0",
-    "rescript": "^10.1.4",
+    "rescript": "^11.0.0-alpha.6",
```
Comment (r17x) [inline package.json]: Maybe want to try beta 👀 [11.0.0-beta.2](https://github.com/rescript-lang/rescript-compiler/releases/tag/v11.0.0-beta.2)

Comment (jihchi) [reply package.json]: Upgraded in https://github.com/jihchi/vitejs-template-react-rescript/pull/18/commits/a86cde10f89b920473eff7fe0fc94ee7e8f74f74

Comment (r17x) [reply package.json]: @jihchi we can use `^11.0.1`

**Comment by jihchi** [comment]
Closing this PR in favor of #22

**Pull request closed without merging**